vortex-fastlanes/src/bitpacking/compress.rs

```diff
@@ -186,20 +185,34 @@ pub fn unpack(array: &BitPackedArray) -> VortexResult<PrimitiveArray> {
             array.validity(),
         ),
         _ => panic!("Unsupported ptype {:?}", ptype),
-    }
-    .into_array();
+    };
 
     // Cast to signed if necessary
-    // TODO(ngates): do this more efficiently since we know it's a safe cast. unchecked_cast maybe?
     if ptype.is_signed_int() {
-        unpacked = cast(&unpacked, &ptype.into())?
+        unpacked = reinterpret_cast(unpacked, ptype);
     }
 
     if let Some(patches) = array.patches() {
-        unpacked = patch(unpacked.as_ref(), patches)?;
+        patch_owned(unpacked, patches)
+    } else {
+        Ok(unpacked)
     }
+}
+
+fn reinterpret_cast(array: PrimitiveArray, ptype: PType) -> PrimitiveArray {
```

Comment:
Shouldn't this be how cast is implemented anyway for primitive array?

Reply:
you can't do this for any ptype cast, this only works for unsigned to signed casts. I can move it there

Reply:
ok, I thought we wouldn't want it there since this doesn't validate values. In this place we know the values can be reinterpreted and you wouldn't even get a null, just a garbage value